pinot-common/src/main/java/org/apache/pinot/sql/parsers/CalciteSqlParser.java

```diff
@@ -708,12 +708,12 @@ private static Expression compileFunctionExpression(SqlBasicCall functionNode) {
       case OTHER:
       case OTHER_FUNCTION:
       case DOT:
-        functionName = functionNode.getOperator().getName().toUpperCase();
+        functionName = canonicalize(functionNode.getOperator().getName());
```

Comment:
The broker allocates a lot in this area, do you think it’s worth mapping from common forms to canonical form to save some string allocs and copies?

Reply:
It requires function information to maintain this map, it might also be hard to derive the camel case, snake case etc from the original function name. To reduce the non-necessary allocation, I think we can use lower case here as we use lower case to canonicalize the function name in other places. Then if we find this is the next bottleneck, we can consider adding this map then.

Reply:
It’s not really a case of it being a bottleneck but a death by 1000 cuts in this area, no single improvement will be transformative, but processing sql allocates a lot which leads to a lot of CPU time spent on GC, if not much latency impact

Reply:
I understand the concern of allocating extra strings here, but canonicalizing the function name to lower case here won't make things worse because it can avoid the extra allocation in multiple places such as converting to lower case in `FunctionContext` and creating `AggregationFunction` and `TransformFunction`.
To further optimize this, we could potentially put all supported aggregation/transform/SqlKind into a map, but not sure if it can really save much and worth maintaining the map and code complexity as the map size could be large if we put all these keys. I'd prefer keeping it this way for now and optimize it in a separate PR if necessary.

Reply:
Let’s revisit this at a later date